Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1629,6 +1630,229 @@
     UNLOCK_CAIRO();
 }
 
+void CGContextCairo::_CGContextDrawStrokedPath(cairo_path_t* flatPath, int i, int prev, int total, CGFloat lineWidth) {
+    cairo_path_data_t* data = &flatPath->data[i];
+    int nextIndex = i + flatPath->data[i].header.length;
+    bool moreData = nextIndex < total;
+    // Try to get next and previous types.
+    cairo_path_data_type_t nextType = CAIRO_PATH_CLOSE_PATH;
+    cairo_path_data_type_t prevType = CAIRO_PATH_CLOSE_PATH;
+    if (moreData) {
+        nextType = flatPath->data[nextIndex].header.type;
+    }
+    if (i > 0) {
+        prevType = flatPath->data[prev].header.type;
+    }
+    switch (data->header.type) {
+        case CAIRO_PATH_MOVE_TO: {
+            // Don't do a move here, it'll move to the center of our stroke.
+            // Save the current position for later. This will help us close the current sub path
+            CGPoint lastPos = curPathPosition;
+            // Set the current position
+            curPathPosition.x = data[1].point.x;
+            curPathPosition.y = data[1].point.y;
+
+            // If there's more data process it
+            if (moreData) {
+                _CGContextDrawStrokedPath(flatPath, nextIndex, i, total, lineWidth);
```

Comment:
Any way I think about this it requires 2 loops because of the way I need to traverse the original path. Then I'd need to either redo or save any calculations needed in both loops. Is that a fair tradeoff?